non-wheel/httpsrv/httpsrv.py

```diff
@@ -0,0 +1,429 @@
+
+; Where to place socket files
+IpcDir /www/{}/logs
+'''.format(name)
+
+
+def get_fastcgi_conf_twn(name):
+    return '''; Static PHP servers for default user
+Server type="application/x-httpd-php" CommandLine="/usr/local/zendphp7/bin/php-cgi.bin" StartProcesses="1" SetEnv="LIBPATH=/usr/local/zendphp7/lib" SetEnv="PHPRC=/usr/local/zendphp7/etc/" SetEnv="PHP_FCGI_CHILDREN=10" SetEnv="PHP_FCGI_MAX_REQUESTS=0" ConnectionTimeout="30" RequestTimeout="60" SetEnv="CCSID=1208" SetEnv="LANG=C" SetEnv="INSTALLATION_UID=100313092679" SetEnv="LDR_CNTRL=MAXDATA=0x40000000" SetEnv="ZEND_TMPDIR=/usr/local/zendphp7/tmp" SetEnv="TZ=<EST>-5"
+
+; Where to place socket files
+IpcDir /www/{}/logs    
+'''.format(name)
+
+
+def get_fastcgi_dynamic_conf(name):
+    return '''; Static PHP servers for default user
+DynamicServer type="application/x-httpd-php" MinProcesses=5 MaxProcesses=100 CommandLine="/usr/local/zendphp7/bin/php-cgi.bin" SetEnv="LIBPATH=/usr/local/zendphp7/lib" SetEnv="PHPRC=/usr/local/zendphp7/etc/" SetEnv="PHP_FCGI_CHILDREN=1" SetEnv="PHP_FCGI_MAX_REQUESTS=0" ConnectionTimeout="60" RequestTimeout="60" SetEnv="CCSID=1208" SetEnv="LANG=en_US" SetEnv="INSTALLATION_UID=100313092679" SetEnv="LDR_CNTRL=MAXDATA=0x40000000" SetEnv="TZ=<EST>5<EDT>,M3.2.0,M11.1.0"
+
+; Where to place socket files
+IpcDir /www/{}/logs
+
+;Minimum and Maximum of dynamic servers
+MinDynamicServers 5
+MaxDynamicServers 100
+
+
+;Usage of this configuration requires follwoing PASE and DG1 PTFs
+;PASE PTFs
+;V5R4: SI43218
+;V6R1: SI43243
+;V7R1: SI43244
+
+;DG1 PTFs
+;V5R4: SI43221
+;V6R1: SI43224
+;V7R1: SI43222
+'''.format(name)
+
+
+def get_fastcgi_http_add_conf(name):
+    return '''Include /www/{}/conf/apache-sites 
+'''.format(name)
+
+
+def get_apache_conf(name, port):
+    return '''# Apache Default server configuration
+LoadModule proxy_module /QSYS.LIB/QHTTPSVR.LIB/QZSRCORE.SRVPGM
+LoadModule proxy_http_module /QSYS.LIB/QHTTPSVR.LIB/QZSRCORE.SRVPGM
+LoadModule proxy_connect_module /QSYS.LIB/QHTTPSVR.LIB/QZSRCORE.SRVPGM
+LoadModule proxy_ftp_module /QSYS.LIB/QHTTPSVR.LIB/QZSRCORE.SRVPGM
+LoadModule zend_enabler_module /QSYS.LIB/QHTTPSVR.LIB/QZFAST.SRVPGM
+
+# zend fastcgi
+AddType application/x-httpd-php .php
+AddHandler fastcgi-script .php
+
+# General setup directives
+Listen *:{0}
+HotBackup Off
+HostNameLookups Off
+UseCanonicalName On
+TimeOut 30000
+KeepAlive On
+KeepAliveTimeout 
+DocumentRoot /www/{1}/htdocs
+AddLanguage en .en
+
+# protection (Basic)
+<Directory />
+   Order Deny,Allow 
+   Deny From all     
+</Directory>
+
+<Directory /www/{1}/htdocs>
+  Options FollowSymLinks 
+  order allow,deny
+  allow from all
+  AllowOverride all

```

Comment:
The `order allow,deny` and `order deny,allow` don't seem to work on IBM i 7.2 for me. I had to change them to `Require all granted` and `Require all denied` instead.

If you want to support 7.1 and 7.2 syntax you can query uname:

```python
import os
un = os.uname()
version = (int(un.version), int(un.release))
if version > (7,1):
....
```

Reply:
What do you think of the latest commit? Mine keeps timing out still.

Reply:
Once I added the KeepAliveTimeout, the server started and then once I changed the Apache 2.2 permission syntax to Apache 2.4 syntax it worked for me.